jax/_src/dispatch.py

```diff
@@ -591,11 +591,18 @@ def _device_put_impl(
     device: Optional[Union[Device, Sharding]] = None,
     src: Optional[Union[Device, Sharding]] = None):
   from jax._src import array
+  if not isinstance(device, (Device, Sharding, type(None))):
```

Comment:
Why not do this check in api.py before we bind?

Reply:
We could! Actually usually we put these in the abstract eval rule only, so that we don't always do it when we're tracing, in principle to make eager mode faster. But I saw other checks in the impl rule so I just added it there.

Reply:
I am pretty sure a certain internal project will fail on this isinstance check :)